clang/lib/AST/ASTContext.cpp

```diff
@@ -2201,6 +2207,9 @@ TypeInfo ASTContext::getTypeInfoImpl(const Type *T) const {
     Align = 8;                                                                 \
     break;
 #include "clang/Basic/WebAssemblyReferenceTypes.def"
+    case BuiltinType::AMDGPUBufferRsrc:
+      Width = 0;
+      Align = 256;
```

Comment:
This has to be power of 2 so 160 -> 256.

Reply:
If we expose this as a pointer there are a lot of sharp edges we need to fix. If we expose it as an opaque type for builtins, the broken cases are hidden from end users.

Reply:
Based on our data layout, it looks like AS8 pointer is 128 bit and AS7 is 160 bit. The alignment here should be 128.